<tt>
&lt;html&gt;&lt;head&gt;&lt;/head&gt;&lt;body&nbsp;style=&quot;word-wrap:&nbsp;break-word;&nbsp;-webkit-nbsp-mode:&nbsp;space;&nbsp;-webkit-line-break:&nbsp;after-white-space;&nbsp;color:&nbsp;rgb(0,&nbsp;0,&nbsp;0);&nbsp;font-size:&nbsp;14px;&nbsp;font-family:&nbsp;Calibri,&nbsp;sans-serif;&nbsp;&quot;&gt;&lt;div&gt;I'm&nbsp;encountering&nbsp;a&nbsp;strange&nbsp;situation&nbsp;that&nbsp;I&nbsp;don't&nbsp;think&nbsp;has&nbsp;been&nbsp;covered&nbsp;by&nbsp;other&nbsp;threads.&lt;/div&gt;&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;div&gt;I'm&nbsp;running&nbsp;RabbitMQ&nbsp;2.7.0&nbsp;on&nbsp;Ubuntu&nbsp;10.04&nbsp;LTS&nbsp;with&nbsp;the&nbsp;stock&nbsp;PIKA&nbsp;0.9.5&nbsp;client,&nbsp;and&nbsp;taking&nbsp;advantage&nbsp;of&nbsp;mirrored&nbsp;queues.&lt;/div&gt;&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;div&gt;My&nbsp;scenario:&lt;/div&gt;&lt;ul&gt;&lt;li&gt;I&nbsp;have&nbsp;two&nbsp;apps&nbsp;that&nbsp;ping-pong&nbsp;messages&nbsp;to&nbsp;each&nbsp;other&nbsp;using&nbsp;two&nbsp;queues&nbsp;(a&nbsp;pseudo-RPC&nbsp;type&nbsp;of&nbsp;thing.)&lt;/li&gt;&lt;li&gt;The&nbsp;apps&nbsp;talk&nbsp;to&nbsp;RabbitMQ&nbsp;via&nbsp;a&nbsp;keepalived&nbsp;monitored&nbsp;Virtual&nbsp;IP.&lt;/li&gt;&lt;li&gt;Before&nbsp;starting&nbsp;the&nbsp;apps,&nbsp;I&nbsp;predeclare&nbsp;both&nbsp;queues.&lt;/li&gt;&lt;li&gt;In&nbsp;the&nbsp;RabbitMQ&nbsp;management&nbsp;Web&nbsp;UI,&nbsp;I&nbsp;see&nbsp;both&nbsp;nodes&nbsp;and&nbsp;the&nbsp;Queues&nbsp;tab&nbsp;shows&nbsp;both&nbsp;queues&nbsp;as&nbsp;&quot;HA&nbsp;D&quot;,&nbsp;so&nbsp;I&nbsp;believe&nbsp;they're&nbsp;mirrored.&lt;/li&gt;&lt;li&gt;Start&nbsp;up&nbsp;the&nbsp;two&nbsp;ping-ponging&nbsp;apps,&nbsp;let&nbsp;run&nbsp;for&nbsp;few&nbsp;seconds.&nbsp;Note&nbsp;messages&nbsp;flowing&nbsp;back&nbsp;and&nbsp;forth.&lt;/li&gt;&lt;li&gt;Stop&nbsp;the&nbsp;primary&nbsp;node&nbsp;(rabbitmqctl&nbsp;stop)&lt;/li&gt;&lt;li&gt;Keepalived&nbsp;fails&nbsp;over&nbsp;the&nbsp;Virtual&nbsp;IP&lt;/li&gt;&lt;li&gt;After&nbsp;a&nbsp;few&nbsp;seconds&nbsp;I&nbsp;see&nbsp;one&nbsp;of&nbsp;the&nbsp;apps&nbsp;resume&nbsp;processing&nbsp;messages&nbsp;as&nbsp;expected.&lt;/li&gt;&lt;li&gt;Shortly&nbsp;thereafter&nbsp;the&nbsp;other&nbsp;app&nbsp;gets&nbsp;an&nbsp;exception&nbsp;from&nbsp;channel.basic_get(),&nbsp;which&nbsp;is&nbsp;the&nbsp;404&nbsp;&quot;no&nbsp;queue&nbsp;found&quot;&nbsp;exception.&lt;/li&gt;&lt;li&gt;I&nbsp;don't&nbsp;believe&nbsp;it's&nbsp;a&nbsp;connection&nbsp;error,&nbsp;as&nbsp;during&nbsp;my&nbsp;queue&nbsp;reads,&nbsp;my&nbsp;logic&nbsp;catches&nbsp;AMQPConnectionErrors&nbsp;exceptions&nbsp;and&nbsp;retries&nbsp;after&nbsp;a&nbsp;short&nbsp;wait.&lt;/li&gt;&lt;/ul&gt;&lt;div&gt;I've&nbsp;looked&nbsp;at&nbsp;my&nbsp;code&nbsp;carefully,&nbsp;and&nbsp;there's&nbsp;no&nbsp;calls&nbsp;to&nbsp;queue_declare&nbsp;other&nbsp;than&nbsp;the&nbsp;original&nbsp;creation&nbsp;(item&nbsp;3,&nbsp;above.)&lt;/div&gt;&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;div&gt;Any&nbsp;thoughts&nbsp;on&nbsp;what&nbsp;could&nbsp;be&nbsp;wrong?&lt;/div&gt;&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;div&gt;Thanks&nbsp;much,&lt;/div&gt;&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;div&gt;Matt&lt;/div&gt;&lt;div&gt;&lt;br&gt;&lt;/div&gt;&lt;/body&gt;&lt;/html&gt;<br>

</tt>
